Fight.java

```diff
@@ -1,43 +1,143 @@
+    animalHealth1.setBackground(Color.darkGray);
+    animalHealth1.setBounds(10, 10, 500, 40);
+    animalHealth1.setValue(10000);
+    animalHealth1.setStringPainted(true);
+
+    JLabel animalOne = new JLabel(one.name().toUpperCase());
+    animalOne.setBounds(515,10,85,40);
+    animalOne.setFont(new Font(null, Font.BOLD, 20));
+
+    animalHealth2 = new JProgressBar(0, 10000);
+    animalHealth2.setForeground(Color.red);
+    animalHealth2.setBackground(Color.darkGray);
+    animalHealth2.setBounds(10,100,500,40);
+    animalHealth2.setValue(10000);
+    animalHealth2.setStringPainted(true);
+
+    JLabel animalTwo = new JLabel(two.name().toUpperCase());
+    animalTwo.setBounds(515,100,85,40);
+    animalTwo.setFont(new Font(null, Font.BOLD, 20));
+
+    JLabel winner = new JLabel();
+    winner.setBounds(10,200,400,100);
+    winner.setFont(new Font(null, Font.BOLD, 30));
+
+    backButton = new JButton("Back");
+    backButton.setBounds(595, 205, 95, 25);
+    backButton.addActionListener(new Fight());
+
+    exitButton = new JButton("Exit");
+    exitButton.setBounds(595, 235, 95, 25);
+    exitButton.addActionListener(new Fight());
+
+    rematchButton = new JButton("Rematch");
+    rematchButton.setBounds(595, 175, 95, 25);
+    rematchButton.addActionListener(new Fight());
+
+
+    game.add(animalHealth1);
+    game.add(animalOne);
+    game.add(animalHealth2);
+    game.add(animalTwo);
+    game.add(winner);
+    game.add(backButton);
+    game.add(exitButton);
+    game.add(rematchButton);
+
+    frame.setVisible(true);
+    int delay = 100;
+    ActionListener taskPerformer = new ActionListener() {
+      public void actionPerformed(ActionEvent e) {
+        int health1 = animalHealth1.getValue();
+        int health2 = animalHealth2.getValue();
+        if (animalHealth1.getValue() > 0 && animalHealth2.getValue() > 0) {
+          health1 -= attack1(one, two, currentEnvironment);
+          animalHealth1.setValue(health1);
+          animalHealth1.setString(health1+"/"+10000+" HP");
+          if(animalHealth1.getValue() > 0) {
+            health2 -= attack1(two, one, currentEnvironment);
+            animalHealth2.setValue(health2);
+            animalHealth2.setString(health2+"/"+10000+" HP");
+            if(animalHealth2.getValue() <= 0) {
+              animalHealth2.setString(0+"/"+10000+" HP");
+              winner.setText("The Winner Is: " + one.name());
+              ((Timer)(e.getSource())).stop();
+            }
+          } else {
+            animalHealth1.setString(0+"/"+10000+" HP");
+            ((Timer)(e.getSource())).stop();
+            winner.setText("The Winner Is: " + two.name());
+          }
+        } 
       }
-      System.out.println(winner);
-      return winner;
+    };
+    timer = new Timer(delay, taskPerformer);
+    timer.setRepeats(true);
+    if (animalHealth1.getValue() > 0) {
+      timer.start();
+    }
+  }
+
+  public void actionPerformed(ActionEvent e) {
```

Comment:
You should named these even handlers more precisely to what event they are handling rather than the generic `actionPerformed`.  It will make you code way easier to follow and maintain.